src/commands/commerce/store/quickstart/setup.ts

```diff
@@ -478,32 +496,37 @@ export class StoreQuickstartSetup extends SfdxCommand {
     }
 
     private updateSelfRegProfile(): void {
+        this.ux.log(msgs.getMessage('quickstart.setup.updateSelfRegProfile'));
         const networkMetaFile = `${this.storeDir}/experience-bundle-package/unpackaged/networks/${
             this.varargs['communityNetworkName'] as string
         }.network`;
-        let data = fs
-            .readFileSync(networkMetaFile)
-            .toString()
-            .replace(/<selfRegProfile>.*<\/selfRegProfile>/g, '')
-            .replace(
-                '</Network>',
-                `    <selfRegProfile>Buyer_User_Profile_From_QuickStart${
-                    StoreQuickstartSetup.getStoreType(
-                        this.org.getUsername(),
-                        this.flags,
-                        this.ux,
-                        this.logger
-                    ).toLowerCase() === 'b2b'
-                        ? '_B2B'
-                        : ''
-                }</selfRegProfile>\n</Network>`
-            );
+
+        let data = fs.readFileSync(networkMetaFile).toString();
+
+        if (StoreQuickstartSetup.getStoreType(this.org.getUsername(), this.flags, this.ux, this.logger) === 'B2C') {
+            data = data
+                .replace(/<selfRegProfile>.*<\/selfRegProfile>/g, '')
+                .replace(
+                    '</Network>',
+                    `    <selfRegProfile>Buyer_User_Profile_From_QuickStart${
```

Comment:
if it is applicable only for B2C, we can remove this condition

Reply:
the `updateSelfRegProfile` function gets called for both B2B and B2C stores (see line [1003](https://github.com/sandragolden/commerce-on-lightning/blob/sgolden/W-12277635-url-path-prefix/src/commands/commerce/store/quickstart/setup.ts#L1003)), but this section should only execute for B2C stores

Reply:
ok, then we can remove line from 512 to 519

Reply:
all set, thanks!